### PR TITLE
Fix video search

### LINKED ISSUE #438: Exception "playlist response is broken" is thrown when searching for videos
Exception thrown:

at "YoutubeClient.Search.GetVideosAsync("Blue Magic Guess Who").BufferAsync()"

Message: 
Playlist response is broken. In most cases, this error indicates that the problem is on YouTube's side and this is not a bug in the library. To resolve this error, please wait some time and try again. If this issue persists, please report it on the project's GitHub page.

StackTrace:
   bei YoutubeExplode.ReverseEngineering.Responses.PlaylistResponse.Parse(String raw)
   bei YoutubeExplode.ReverseEngineering.Responses.PlaylistResponse.<>c__DisplayClass12_0.<<GetSearchResultsAsync>b__0>d.MoveNext()
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei YoutubeExplode.Internal.Retry.<WrapAsync>d__1`1.MoveNext()
   bei YoutubeExplode.Internal.Retry.<WrapAsync>d__1`1.MoveNext()
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei YoutubeExplode.ReverseEngineering.Responses.PlaylistResponse.<GetSearchResultsAsync>d__12.MoveNext()
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei YoutubeExplode.Search.SearchClient.<GetVideosAsync>d__2.MoveNext()
   bei System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.GetResult(Int16 token)
   bei YoutubeExplode.Search.SearchClient.<GetVideosAsync>d__2.System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult(Int16 token)
   bei YoutubeExplode.Internal.Extensions.AsyncEnumerableExtensions.<ToListAsync>d__1`1.MoveNext()
   bei YoutubeExplode.Internal.Extensions.AsyncEnumerableExtensions.<ToListAsync>d__1`1.MoveNext()
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei YoutubeExplode.AccessibilityExtensions.<BufferAsync>d__0.MoveNext()
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei System.Threading.Tasks.ValueTask`1.get_Result()
   bei System.Runtime.CompilerServices.ValueTaskAwaiter`1.GetResult()

### PR DESCRIPTION
**Pull request by Unreal852**
Closes #438

### CODE REVIEW

File: YoutubeExplode/YoutubeClient.cs [hunk 1]
```diff
@@ -75,6 +75,8 @@ public partial class YoutubeClient
 
             httpClient.DefaultRequestHeaders.Add("User-Agent",
                 "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.163 Safari/537.36");
+            httpClient.DefaultRequestHeaders.Add("x-youtube-client-name", "56");
+            httpClient.DefaultRequestHeaders.Add("x-youtube-client-version", "20200911");
```
Comment (Hexer10) [inline YoutubeExplode/YoutubeClient.cs:79]: These headers should be added only to the request that calls the search api, if you put them in the other requests they will fail; try to run the unit tests.

**Review by Hexer10 — CHANGES_REQUESTED** [review]

**Comment by Unreal852** [comment]
whoops didn't think about that, i fixed it. Altrought i still have issues with Closed Captions and Streams.

**Comment by Hexer10** [comment]
Well... Yeah that the video used in the unit tests is now private: https://www.youtube.com/watch?v=_QdPW8JrYzQ , if you want you can check my Dart version of this library and update the tests from there


https://github.com/Hexer10/youtube_explode_dart/blob/master/test/closed_caption_test.dart#L15-L41

**Comment by Tyrrrz** [comment]
Thanks!

**Pull request merged**